### ISSUE TITLE
AnnotationLayer links misaligned

### ISSUE THREAD

**Issue by Mrozuu11** [wojtekmaj/react-pdf#1056]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [ ] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

The links contained in my pdf file are misaligned while using the react-pdf. I can see a bright yellow box and a cursor pointer indicating the link but it's out of place (I tried showing it on a screenshot below)

![image](https://user-images.githubusercontent.com/78096071/192565566-8f9853b1-15f9-4ece-ad11-cdacc8bdb2dd.png)

### Steps to reproduce

Create a NextJs project
Copy the code below
Attach a pdf file

### Expected behavior

Annotations to be aligned with hyperlink text

### Actual behavior

Annotations being misaligned

### Additional information

My code is as follows:

```
import { Document, DocumentProps, Page, pdfjs } from "react-pdf";
import "react-pdf/dist/esm/Page/AnnotationLayer.css";
pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.min.js`;

interface galleryProps {
  isOpen: boolean;
  handleClose: () => void;
}
const options = {
  cMapUrl: "cmaps/",
  cMapPacked: true,
  standardFontDataUrl: "standard_fonts/",
};
type LoadCallback = Required<DocumentProps>["onLoadSuccess"];

 const [numPages, setNumPages] = useState(0);
  const [pageNumber, setPageNumber] = useState(1);
  const onLoad: LoadCallback = useCallback((pdf) => {
    setNumPages(pdf.numPages);
  }, []);
  const handleNextPage = () => {
    pageNumber < 2
      ? setPageNumber(pageNumber + 1)
      : setPageNumber(pageNumber - 1);
  };
  const setDefaultPage = () => {
    setTimeout(() => setPageNumber(1), 500);
  };
  const handlers = useSwipeable({
    onSwipedDown: () => handleNextPage(),
    onSwipedUp: () => handleNextPage(),
  });
  return (
    <section
      className={`${styles.container} ${isOpen && styles.open}`}
      onWheel={handleNextPage}
      {...handlers}
    >
      <button
        className={styles["close-btn"]}
        aria-label="close-cv-gallery"
        onClick={() => {
          handleClose();
          setDefaultPage();
        }}
      >
        <CloseIcon fontSize="large" className={styles.icon} />
      </button>

      <Document
        file="/CV/mm-cv.pdf"
        onLoadSuccess={onLoad}
        externalLinkTarget="_blank"
        options={options}
        className={styles.document}
      >
        <Page
          pageNumber={pageNumber}
          renderAnnotationLayer={true}
          renderTextLayer={false}
        />
        <button
          onClick={handleNextPage}
          aria-label="cv-next-page"
          type="button"
          className={styles["next-btn"]}
        >
          <ArrowCircleDownOutlinedIcon
            fontSize="large"
            className={styles.icon}
          />
        </button>
      </Document>

      <p>
        Page {pageNumber} of {numPages}
      </p>
    </section>
  );
}
```

### Environment

- **Browser (if applicable)**: Brave v1.43 (Chromium 105)
- **React-PDF version**: 5.7.2
- **React version**: 18.2
- **NextJS 12.2

**Comment by Ahmed-Mouzoune** [comment]
I met the same problem yesterday, I modified the css of the library by hand and it works on my side you can always try ^^ :
`.react-pdf__Document {
    display: flex;
    .react-pdf__Page {
        .react-pdf__Page__textContent {top: 0 !important; left: 0 !important; transform: none !important;}
        .react-pdf__Page__annotations {position: absolute; top: 0 !important; left: 0 !important; transform: none !important;
            section {
                position: absolute;
                top: 0px;
                left: 0px;
            }
            section > a {display: block; width: 100%; height: 100%;}
        }
    }
}`

**Comment by Mrozuu11** [comment]
@Ahmed-Mouzoune Thanks for the tip. Unfortunately it doesn't solve my issue but just puts all the links at the top of the Document, I did think about changing the position properties but it wouldn't work with the responsive design in my case :/

**Comment by Ahmed-Mouzoune** [comment]
Indeed it works only if you use the width property of the document :/, you can always try to switch to React 17.0 by changing the react and react-dom dependencies in the package.json file like this :
`"react": "^17.0.2",  
"react-dom": "^17.0.2",`
 Then `npm install` and see if it works

**Comment by CallumJHays** [comment]
I fixed this by overriding the styles:

```tsx
.react-pdf__Page__textContent {
  top: 0 !important;
  left: 0 !important;
  transform: none !important;
}
```

This does the trick for me, although text after tabs doesn't always render at the right place in the text DOM.

**Comment by wojtekmaj** [comment]
Check out 6.0.0 beta - rendering these layers were completely revamped in this version.

**Comment by Eivlys** [comment]
try to upgrade to version 6 but I got  this building problem [https://github.com/wojtekmaj/react-pdf/issues/1043](url) 

 manually import the css and set `externalLinkTarget ` to `_blank` can also fix this in v5.7.2
 ```
import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
...

  <Page
      key={`page_${index + 1}`}
      pageNumber={index + 1}
      width={375}
      loading="Loading Page..."
      renderAnnotationLayer={true}
      renderTextLayer={false}
      externalLinkTarget="_blank"
  />
```
